lib/trip_dispatcher.rb

```diff
@@ -19,20 +25,73 @@ def find_passenger(id)
       return @passengers.find { |passenger| passenger.id == id }
     end
 
+    def find_driver(id)
+      Driver.validate_id(id)
+      return @drivers.find { |driver| driver.id == id }
+    end
+
     def inspect
       # Make puts output more useful
       return "#<#{self.class.name}:0x#{object_id.to_s(16)} \
-              #{trips.count} trips, \
-              #{drivers.count} drivers, \
-              #{passengers.count} passengers>"
+      #{trips.count} trips, \
+      #{drivers.count} drivers, \
+      #{passengers.count} passengers>"
+    end
+
+    def request_trip(passenger_id)
+
+      selected_driver = nil
+
+      @drivers.each do |driver_instance|
+        if driver_instance.status == :AVAILABLE
+          selected_driver = driver_instance
+          break
+        end
+      end
```

Comment:
Great work working through this logic! Consider breaking out this functionality into a helper method.